data_structures/node_based_graph.hpp

```diff
@@ -80,8 +81,9 @@ NodeBasedDynamicGraphFromEdges(int number_of_nodes, const std::vector<NodeBasedE
     auto edges_list = directedEdgesFromCompressed<NodeBasedDynamicGraph::InputEdge>(input_edge_list,
         [](NodeBasedDynamicGraph::InputEdge& output_edge, const NodeBasedEdge& input_edge)
         {
-            output_edge.data.distance = static_cast<int>(input_edge.weight);
-            BOOST_ASSERT(output_edge.data.distance > 0);
+            output_edge.data.weight = static_cast<int>(input_edge.weight);
+            output_edge.data.duration = static_cast<int>(input_edge.duration);
+            BOOST_ASSERT(output_edge.data.weight > 0);
```

Comment:
Are those casts safe? Those are narrowing conversions.

Discussion: maybe we should think about either providing a `safe_cast` or using `boost::numeric_cast` --- not just for these two casts but globally.